java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryProvider.java

```diff
@@ -79,15 +84,24 @@ public final Set<Long> getDictionaryIds() {
     }
 
     @Override
-    public Dictionary lookup(long id) {
+    public BaseDictionary lookup(long id) {
       return map.get(id);
     }
 
     @Override
     public void close() {
-      for (Dictionary dictionary : map.values()) {
+      for (BaseDictionary dictionary : map.values()) {
         dictionary.getVector().close();
       }
     }
+
+    @Override
+    public void resetBatchedDictionaries() {
```

Comment:
It is strange that only batched dictionaries need to be reset.

Reply:
Should I just make a generic reset API with the `Dictionary` having a no-op reset?

Reply:
I think so. It would be more consistent.